### PR TITLE
adding docs on managing cluster wide storage controls

### PR DESCRIPTION
**Pull request by xphyr**
The following PR is a copy and update of a blog post I originally published on xphyr.net on a way to manage storage at a cluster level.  I have updated the documentation per the contribution guide, and removed any references to the file being a blog post.

@mulbc 

Signed-off-by: Mark DeNeve <markd@xphyr.net>

### CODE REVIEW

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+Annotations:	<none>
+Namespace Selector: ["myproject" "myproject2"]
+Label Selector:
+AnnotationSelector: clusterstoragequota=enabled
+Resource            Used	Hard
+--------            ----	----
+requests.storage    25Gi	40Gi
+----
+
+Note that the used storage for the cluster has increased by 20Gi. To validate that the *ClusterResourceQuota* is enforcing our quota across multiple projects, create one more pvc file called _storageclaim3.yaml_ and change the storage request to 20Gi. This request is within the project level quota we have remaining but will exceed the maximum amount of cluster storage we want to allocate.
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:222]: What project does storageclaim3.yaml get created in? I think `-n my project`? Hard to follow logic if not clear what project PVCs created in.

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+      ocs-storagecluster-cephfs.storageclass.storage.k8s.io/requests.storage: "80Gi"
+      ocs-storagecluster-ceph-rbd.storageclass.storage.k8s.io/requests.storage: "0Gi"
+  selector:
+    annotations:
+      clusterstoragequota: enabled
+----
+
+The YAML above creates a cluster level quota for storage that does the following:
+
+* Allow no more than 100Gi of storage to be assigned in your cluster (perhaps your backup service has a max capacity of 100Gi)
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:344]: Not sure what `Allow no more than 100Gi of storage to be assigned in your cluster` means? How is this storage consumed? The underlining storage available would be 100Gi + 40Gi + 80Gi?

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+thin.storageclass.storage.k8s.io/requests.storage	    10Gi	40Gi
+----
+
+Feel free to jump back to the <<exercise-the-quotas,Exercise the quotas>> section and target the additional storage classes you created to see how they work.
+
+== Summary
+
+By combining OpenShift *Project Templates*, and *ClusterResouceQuotas* along with project quotas OpenShift cluster administrators can take back control and manage storage allocated within their cluster. Remember that you will need to retrofit all existing projects that use *PVCs* to have the annotation on the project following the steps in the <<create-and-annotate-projects,Create and Annotate Projects>> section of this document. Use caution when applying annotations to existing projects to ensure that you do not adversely effect applications deployed in these projects. The concepts that are shown here apply to any object type that can have a quota applied.
+
+== References
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:377]: OCP 4.8 will be GA very soon, change to 4.8 references?

**Review by netzzer — CHANGES_REQUESTED** [review]

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc (hunk 3 above)
Comment (xphyr) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:377]: @netzzer this is a great question, and leads to a secondary question... should I use "OCS" or "ODF" in this document?  I am more than happy to update the references to 4.8 and use "ODF" if we believe this is the right way to do it.  I will wait to hear back from you on what the right answer is.

File: training/modules/ocs4/nav.adoc [hunk 4]
```diff
@@ -9,3 +9,4 @@
 * xref:ocs4-encryption.adoc[External KMS Encryption]
 * xref:ocs4-cluster-downsize.adoc[Downsize existing OCS cluster]
 * xref:ocs4-enable-rgw.adoc[Use RGW in OCS deployment]
+* xref:ocs4-cluster-storage-quotas.adoc[Cluster Wide Storage Management]
```
[the remark this answers is not in the document]
Comment (xphyr) [reply training/modules/ocs4/nav.adoc]: done in latest commit.

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+[source,yaml]
+----
+apiVersion: v1
+kind: ClusterResourceQuota
+metadata:
+  name: totalclusterstorage
+spec:
+  quota:
+    hard:
+      requests.storage: "40Gi"
```
[the remark this answers is not in the document]
Comment (xphyr) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:65]: Please see latest commit for clarification text and advise

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+    requests:
+      storage: 5Gi
+  storageClassName: <storageClassName>
+----
+
+Create the *PVC* in your project `oc create -f storageclaim1.yaml`. Now see how the *PVC* you just created is reflected in both your *Project* and cluster quotas:
+
+[source,shell]
+----
+$ oc create -f storageclaim1.yaml
```
[the remark this answers is not in the document]
Comment (xphyr) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc]: done in latest commit.

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc (hunk 1 above)
Comment (xphyr) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc]: fixed in latest commit.

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc (hunk 2 above)
Comment (xphyr) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc]: I updated the text, changing "Allow" to "Ensure" which I think is much clearer.

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+Created:	2 days ago
+Labels:		<none>
+Annotations:	<none>
+Namespace Selector: ["myproject" "myproject2"]
+Label Selector:
+AnnotationSelector: clusterstoragequota=enabled
+Resource            Used	Hard
+--------            ----	----
+requests.storage    5Gi	40Gi
+$ oc describe quota
```
[the remark this answers is not in the document]
Comment (xphyr) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc]: I have attempted to clarify this with information and description in an earlier section when project quotas are first introduced, as well as updating this section to clarify what the information is that is being shown. Please let me know if this clarifies and covers it.

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc [hunk 8]
```diff
@@ -341,10 +347,10 @@ spec:
 
 The YAML above creates a cluster level quota for storage that does the following:
 
-* Allow no more than 100Gi of storage to be assigned in your cluster (perhaps your backup service has a max capacity of 100Gi)
-* Allow no more than 40Gi of storage to be assigned in your cluster from the "thin" storage class
-* Allow no more than 80Gi of storage to be assigned in your cluster from the "ocs-storagecluster-cephfs" class
-* Do not allow provisioning of any "ocs-storagecluster-ceph-rbd" storage class
+* Ensures no more than 100Gi of storage can be assigned in your cluster (perhaps your backup service has a max capacity of 100Gi)
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:350]: The 100Gi assigned in your cluster is still confusing to me? What OCP resource is being monitored for the 100Gi? Is  it also for a storageclass?

**Review by netzzer — CHANGES_REQUESTED** [review]

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc (hunk 8 above)
Comment (xphyr) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:350]: @netzzer this is storage of any type ...no matter what.  If its a PV, it counts here. The 100Gi is a maximum across **all** the storage classes that are available within your cluster.  If you have 30Gi in cephfs, 50Gi in vmware thin, and 20Gi in ceph-rbd, then you have hit the 100Gi max, and you can not allocate any additional storage, no matter what the class.

Does that make sense?  Can you think of a better way to sum up that line?

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc (hunk 3 above)
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:383]: I would use ODF given the change was made April 1, 2021.

File: training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc (hunk 8 above)
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-cluster-storage-quotas.adoc:350]: After this you have below. This is more than 100Gi.

```
* Ensures no more than 40Gi of storage can be assigned in your cluster from the "thin" storage class
* Ensures no more than 80Gi of storage can be assigned in your cluster from the "ocs-storagecluster-cephfs" class
```

**Review by netzzer — CHANGES_REQUESTED** [review]

**Review by netzzer — APPROVED** [review]
